src/ivectorbin/compute-vad-from-frame-likes.cc

```diff
@@ -0,0 +1,186 @@
+      "e.g.: compute-vad-from-frame-likes --map=label_map.txt\n"
+      "    scp:likes1.scp scp:likes2.scp ark:vad.ark\n";
+
+    ParseOptions po(usage);
+    std::string map_rxfilename;
+    std::string priors_rxfilename;
+
+    po.Register("map", &map_rxfilename, "Table that defines the frame-level"
+    " labels.  For each row, the first field is the zero-based index of the"
+    " input likelihood archive and the second field is the associated"
+    " integer label.");
+
+    po.Register("priors", &priors_rxfilename, "Table that specifies"
+    " log-priors for each class. For each row, the first field is the"
+    " zero-based index of the input likelihood archive and the second"
+    " field is the log-prior.");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() < 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    unordered_map<int32, int32> map;
+    unordered_map<int32, BaseFloat> priors;
+    int32 num_classes = po.NumArgs() - 1;
+    PrepareMap(map_rxfilename, num_classes, &map);
+    PreparePriors(priors_rxfilename, num_classes, &priors);
+
+    SequentialBaseFloatVectorReader first_reader(po.GetArg(1));
+    std::vector<RandomAccessBaseFloatVectorReader *> readers;
+    std::string vad_wspecifier = po.GetArg(po.NumArgs());
+    BaseFloatVectorWriter vad_writer(vad_wspecifier);
+
+    for (int32 i = 2; i < po.NumArgs(); i++) {
+      RandomAccessBaseFloatVectorReader *reader
+        = new RandomAccessBaseFloatVectorReader(po.GetArg(i));
+      readers.push_back(reader);
+    }
+
+    int32 num_done = 0, num_err = 0;
+    for (;!first_reader.Done(); first_reader.Next()) {
+      std::string utt = first_reader.Key();
+      Vector<BaseFloat> like(first_reader.Value());
+      int32 like_dim = like.Dim();
+      std::vector<Vector<BaseFloat> > likes;
+      likes.push_back(like);
+      if (like_dim == 0) {
+        KALDI_WARN << "Empty vector for utterance " << utt;
+        num_err++;
+        continue;
+      }
+      for (int32 i = 0; i < num_classes - 1; i++) {
+        if (!readers[i]->HasKey(utt)) {
+          KALDI_WARN << "No vector for utterance " << utt;
+          num_err++;
+          continue;
+        }
+        Vector<BaseFloat> other_like(readers[i]->Value(utt));
+        if (like_dim != other_like.Dim()) {
+          KALDI_ERR << "Dimension mismatch in input vectors in " << utt
+                    << ": " << like_dim << " vs. " << other_like.Dim();
+        }
+        likes.push_back(other_like);
+      }
+
+      Vector<BaseFloat> vad_result(like_dim);
+      for (int32 i = 0; i < like.Dim(); i++) {
+        int32 max_indx = 0;
+        BaseFloat max_post = likes[0](i) + priors[0];
+        for (int32 j = 0; j < num_classes; j++) {
+          BaseFloat other_post = likes[j](i) + priors[j];
+          if (other_post > max_post) {
+            max_indx = j;
+            max_post = other_post;
+          }
+        }
+        if (map.find(max_indx) == map.end())
```

Comment:
it's better to get into the habit of using the map<...>::const_iterator here, as in:
map<...>::const_iterator iter = map.find(max_indx);
if (iter == map.end()) {
...
} else {
   // use iter->second.
}